src/Avalonia.Controls/Converters/PlatformKeyGestureConverter.cs

```diff
@@ -57,7 +57,7 @@ public static string ToPlatformString(KeyGesture gesture)
             }
             else
             {
-                return gesture.ToString();
+                return ToString(gesture, "Cmd");
```

Comment:
"Win" key would make more sense since it's likely an external keyboard we are dealing with.

Reply:
I went with Cmd, because that's what the gesture.ToString() prints for Meta.  I agree it's almost certainly an external keyboard, they don't all say Win.  The logitech K480 I've been using for testing is "Start | alt opt".

Reply:
Another thing to keep in mind is that this isn't Android specific, it's any runtime that's not nominally Windows, Linux, or OSX. 

Actually looking deeper, it seems using RuntimeInformation.IsPlatform() might no longer be the appropriate way to do this, as a lot of the tests have been added to System.OperatingSystem with more granularity (Browser, Android, iOS, etc).

Reply:
> I went with Cmd, because that's what the gesture.ToString() prints for Meta

That's a good point: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Input/KeyGesture.cs#L130
Possibly it's worth reusing code between PlatformKeyGestureConverter and KeyGesture. Their ToString() and Parse() combination should be compatible and unit tested.

Reply:
> Actually looking deeper, it seems using RuntimeInformation.IsPlatform() might no longer be the appropriate way to do this

In general, yes. We should use OperatingSystem API (or OperatingSystemEx in Avalonia repo). But you probably won't get much out of it, unless you need to add cases for mobile and browser as well. It also won't tell you on which platform browser is running.

Reply:
For the most part the ToString functions are the same.  The biggest difference is the Meta variations in the converter, and using a variety of symbols for the OSXString.  These could probably be moved to KeyGesture.ToString() in some way, and then the converter simply checks the platform and calls as appropriate.  Would it be desirable for the default KeyGesture.ToString() to print user friendly characters rather than enum strings, or would that be considered a breaking change?  I suspect it would.

From what I understand of Key parsing, the default ToString should probably use the Enum value strings, or digits will cause problems.  
So maybe something like?:
```
KeyGesture.ToString(string metaOverride = "cmd", bool prettyPrint = false)
KeyGesture.ToString() => ToString("Cmd", false)
KeyGesture.ToOSXString()
```

Reply:
I took a second look at this issue. You are right, OSX differences in formatted strings make it more difficult.

For the most part, our differences between platform hotkeys are defined by the PlatformHotkeyConfiguration class.
Which also can be redefined by the platform OS as well. Which allows us to avoid RuntimeInformation.IsOSPlatform checks in shared code. 

What do you think about:
1. KeyGesture.ToString() to always use enum values.
2. Add **Key**DisplayName properties to PlatformHotkeyConfiguration or something like this. Can be a dictionary, I guess.
3. KeyGesture.ToFormattedString() that internally uses PlatformHotkeyConfiguration configuration for special key display names. No need for ToOSXString.
4. KeyGesture.Parse/TryParse expects enum values, as it's normally defined in XAML as well. Need to make sure that XAML compiler also uses that. 
5. PlatformKeyGestureConverter to use KeyGesture.ToFormattedString internally.

Now, `ToFormattedString` is not as common in .NET BCL. There are methods named like this in .NET in general, but I couldn't find any in BCL itself. Normally CultureInfo parameter makes all difference for ToString formatting strategy of different types, but we can't use it here. https://www.google.com/search?q=ToFormattedString%20site:learn.microsoft.com

Reply:
If you want to follow the BCL, I think we would need to implement `IFormattable`/`ISpanFormattable` in `KeyGesture`. Then we can use the format string + the format provider to output the gesture as desired.

A new type would provide the modifier key strings (or `PlatformHotkeyConfiguration` as you mentioned, but I think that isn't its responsability) and implement `IFormatProvider`.

Then `KeyGesture` can look for the needed format in `IFormatProvider`, falling back to the current platform and/or invariant values as needed. People are able to provide the format they need.

Sample, using `n` as the invariant format and `p` as platform specific ones:

```csharp
public sealed class KeyModifierFormatInfo(string meta) : IFormatProvider
{
	public string Meta { get; } = meta;
        // TODO: Ctrl, Shift, etc.

	public object? GetFormat(Type? formatType)
		=> formatType == typeof(KeyModifierFormatInfo) ? this : null;

	public static KeyModifierFormatInfo Invariant { get; } = new("Meta");

	public static KeyModifierFormatInfo GetInstance(IFormatProvider? formatProvider)
		=> formatProvider?.GetFormat(typeof(KeyModifierFormatInfo)) as KeyModifierFormatInfo
		   ?? AvaloniaLocator.Current.GetService<KeyModifierFormatInfo>()
		   ?? Invariant;
}

// In KeyGesture:
public override string ToString()
	=> ToString(null, null);

public string ToString(string? format, IFormatProvider? formatProvider)
{
	var formatInfo = format switch
	{
		null or "" or "n" => KeyModifierFormatInfo.Invariant,
		"p" => KeyModifierFormatInfo.GetInstance(formatProvider),
		_ => throw new FormatException("Unknown format specifier");
	};

	// output gesture using formatInfo here
	// ...
}
```

Then it's the matter of documenting the available formats, and its link to `KeyModifierFormatInfo` in the rare event where an user wants to completely customize the output. it might feel a bit over-engineered, but that's the pattern used everywhere by the BCL.

Reply:
Alright, I have a bit of time to work on this now.  Where would the KeyModifierFormatInfo class (or equivalent) go in the Avalonia projects?  Avalonia.Controls or Avalonia.Base are where I'm digging around at the moment, but if there's a better spot please let me know.